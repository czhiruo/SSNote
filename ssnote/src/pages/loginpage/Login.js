import { useState } from "react";
import "./styles.css";
import logo from "../dashboard/assets/SSNote-Logo-gray.png";
import { useNavigate } from "react-router-dom";
import {
  signInWithEmailAndPassword,
  setPersistence,
  browserLocalPersistence,
} from "firebase/auth";
import { Link } from "react-router-dom";
import { auth } from "../../firebase";

function Login() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");

  const navigate = useNavigate();

  //handles logic when users press log in button
  const handleSubmit = (event) => {
    event.preventDefault();

    setPersistence(auth, browserLocalPersistence)
      .then(() => {
        // Sign in with email and password
        return signInWithEmailAndPassword(auth, email, password)
          .then((userCredential) => {
            console.log(userCredential);
            navigate("/dashboard");
          })
          .catch((error) => {
            console.log(error);
            alert("Invalid user");
          });
      })
      .catch((error) => {
        // Handle Errors here.
        const errorCode = error.code;
        const errorMessage = error.message;
        console.log(errorCode);
        console.log(errorMessage);
      });
  };

  //login form display
  const renderForm = (
    <div className="form">
      <form onSubmit={handleSubmit}>
        <div className="input-container">
          <label>Email </label>
          <input
            type="text"
            name="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            required
          />
        </div>
        <div className="input-container">
          <label>Password </label>
          <input
            type="password"
            name="pass"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            required
          />
        </div>
        <Link to={"/forgotpassword"} className="changePassword">
          <u>Forgot Password?</u>
        </Link>

        <br />

        <div className="button-container">
          <input type="submit" value="Log In" />
        </div>

        <br />

        <p id="no-account">Don't have an account?</p>

        <Link to={"/signup"} className="signUp">
          <u>Sign Up Here!</u>
        </Link>
      </form>
    </div>
  );

  return (
    <div className="app">
      <div className="login-form">
        <img className="logo" src={logo} alt="logo" />
        <div className="title">Login</div>
        {renderForm}
      </div>
    </div>
  );
};

export default Login;
