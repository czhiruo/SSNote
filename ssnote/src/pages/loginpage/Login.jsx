import { useState } from "react";
import "./styles.css";
import logo from "./SSNotelogo.jpeg";
import { useNavigate } from "react-router-dom";
import { auth } from "../../firebase";
import { signInWithEmailAndPassword } from "firebase/auth";

const Login = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [isSubmitted, setIsSubmitted] = useState(false);

  const navigate = useNavigate();

  //handles logic when users press log in button
  const handleSubmit = (event) => {
    event.preventDefault();

    signInWithEmailAndPassword(auth, email, password)
      .then((userCredential) => {
        console.log(userCredential);
        navigate("/dashboard")
      })
      .catch((error) => {
        console.log(error);
        alert("Invalid user");
      });
  };

  function handlePasswordChange() {
    alert("Change Password!");
  }

  //link to sign up
  function handleNewAccount() {
    navigate("/signup")
  }

  //login form display
  const renderForm = (
    <div className="form">
      <form onSubmit={handleSubmit}>
        <div className="input-container">
          <label>Email </label>
          <input
            type="text"
            name="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            required
          />
        </div>
        <div className="input-container">
          <label>Password </label>
          <input
            type="password"
            name="pass"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            required
          />
        </div>
        <div className="changePassword" onClick={handlePasswordChange}>
          <u>Forgot Password?</u>
        </div>
        <div className="signUp" onClick={handleNewAccount}>
          <u>Sign Up Here!</u>
        </div>
        <div className="button-container">
          <input type="submit" />
        </div>
      </form>
    </div>
  );

  return (
    <div className="app">
      <div className="login-form">
        <img className="logo" src={logo} alt="logo" />
        <div className="title">Login</div>
        {renderForm}
      </div>
    </div>
  );
};

export default Login;
