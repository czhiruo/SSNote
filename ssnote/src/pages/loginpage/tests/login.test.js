import React from "react";
import { render } from "@testing-library/react";
import Login from "../Login";
import { BrowserRouter } from "react-router-dom";

test("renders Login component correctly", () => {
  const { asFragment } = render(<BrowserRouter><Login /></BrowserRouter>);
  expect(asFragment()).toMatchSnapshot();
});
