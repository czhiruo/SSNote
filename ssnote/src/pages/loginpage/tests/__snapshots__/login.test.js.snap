// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Login component correctly 1`] = `
<DocumentFragment>
  <div
    class="app"
  >
    <div
      class="login-form"
    >
      <img
        alt="logo"
        class="logo"
        src=""
      />
      <div
        class="title"
      >
        Login
      </div>
      <div
        class="form"
      >
        <form>
          <div
            class="input-container"
          >
            <label>
              Email 
            </label>
            <input
              name="email"
              required=""
              type="text"
              value=""
            />
          </div>
          <div
            class="input-container"
          >
            <label>
              Password 
            </label>
            <input
              name="pass"
              required=""
              type="password"
              value=""
            />
          </div>
          <a
            class="changePassword"
            href="/forgotpassword"
          >
            <u>
              Forgot Password?
            </u>
          </a>
          <br />
          <div
            class="button-container"
          >
            <input
              type="submit"
              value="Log In"
            />
          </div>
          <br />
          <p
            id="no-account"
          >
            Don't have an account?
          </p>
          <a
            class="signUp"
            href="/signup"
          >
            <u>
              Sign Up Here!
            </u>
          </a>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
